src/nonce.rs

```diff
@@ -0,0 +1,10 @@
+use rand::{thread_rng, Rng};
+use rand::distributions::Alphanumeric;
+
+pub fn generate_nonce() -> String {
+    thread_rng()
+    .sample_iter(&Alphanumeric)
+    .take(8)
```

Comment:
Just in case, because I was confused as well, the specs provide a bottom limit of 8 characters, but it can actually be more than that.

Reply:
Maybe we add a version that accepts a character count in the future? The TS lib only offers an 8 char nonce which I matched.